### DISCUSSION TITLE
Linux appimage portability issue (.home Folder)

### DISCUSSION THREAD

**Discussion by Mintloid** [flyinghead/flycast#1653] [General]
I'd normally like to have accessible files to configure when it comes to Appimages. Thats where creating folders of the same name as the executable come into play (.config & .home). 

However when I created .home, and right after I launched the emulator, the bios boot wouldn't even load and there was nothing in the folder (.home). I tried relaunching and closing while removing both .config & .home folders and recreating/keeping .config folder only, went to the share folder where most home data flycast has, copied over the files to a fresh new .home folder next to the appimage, and still the .home folder doesn't recognize anything (no boxart, bios boot still won't startup).

I ask this as a data minimalist because I like to mostly keep all apps & their files on bigger drives (2TB HDD is what I have) since my main drive is small (256GB).

### My Specs
OS: Linux Mint 22 Cinnamon
CPU: Intel© Core™ i5-9400F CPU @ 2.90GHz × 6
RAM: 16 GB
GPU: NVIDIA Corporation GeForce GTX 1660 Ti (Driver ver. 550.107.02)
Display Server: X11

**Comment by flyinghead** [comment]
By default, Flycast stores its config file in `~/.config/flycast` and its data files in `~/.local/share/flycast`. The base folders can be changed with the env variables `XDG_CONFIG_HOME` and `XDG_DATA_HOME` respectively (instead of `~/.config` and `~/.local/share`)
I don't know where the `.home` folder comes from but Flycast doesn't use it.
You can move the config and data folders to an external drive and use symbolic links to point to the new locations.

**Reply by xordspar0** [reply]
>  I don't know where the .home folder comes from but Flycast doesn't use it.

AppImage has a convenient feature where if you create a directory like flycast.AppImage.config next to the appimage, when the app starts it will set $XDG_CONFIG_HOME to that directory. Likewise, if you create a directory like flycast.AppImage.home, when the app starts it will set $HOME to that directory. An app that follows the XDG spec should work fine with this with no changes.

I just tried this with v2.4 and it works fine. I would need to see logs or other details to know what happened here.

**Reply by flyinghead** [reply]
I didn't know that. Thanks for the info.
